backend/compact-connect/lambdas/staff-users/data_model/client.py

```diff
@@ -109,61 +109,116 @@ def update_user_permissions(
         logger.info('Updating staff user permissions', user_id=user_id)
 
         update_expression_parts = {'add': [], 'delete': []}
+
+        # DynamoDB does not support both ADD and DELETE on the same String Set in a single UpdateItem call, so we will
+        # split additions and removals into two calls to prevent a conflict.
+        resp = self._handle_user_permission_additions(
+            user_id=user_id,
+            compact=compact,
+            update_expression_parts=update_expression_parts['add'],
+            compact_action_additions=compact_action_additions,
+            jurisdiction_action_additions=jurisdiction_action_additions,
+        )
+        resp = (
+            self._handle_user_permission_removals(
+                user_id=user_id,
+                compact=compact,
+                update_expression_parts=update_expression_parts['delete'],
+                compact_action_removals=compact_action_removals,
+                jurisdiction_action_removals=jurisdiction_action_removals,
+            )
+            or resp
+        )
```

Comment:
Nit: Can we assign these responses to two different variables, and then merge the result together? It would make the code easier to grok.

Reply:
I thought about that, but I don't actually want to merge the results. Each includes a complete copy of the user record and only the last response is valid, so we're just selecting the most recent response to return. Can you think of a more obvious way to present that? I'd be happy to update.